api/tests/mocha/services/generate-xform.spec.js

```diff
@@ -76,22 +84,64 @@ describe('generate-xform service', () => {
       });
     };
 
-    it('generates form and model', () => runTest('simple'));
+    it('generates form and model', () => runTest('simple', spawned));
 
-    it('replaces multimedia src elements', () => runTest('multimedia'));
+    it('replaces multimedia src elements', () => runTest('multimedia', spawned));
 
-    it('correctly replaces models with nested "</root>" - #5971', () => runTest('nested-root'));
+    it('correctly replaces models with nested "</root>" - #5971', () => runTest('nested-root', spawned));
 
-    it('errors if child process errors', done => {
-      runTest('simple', 'some error')
-        .then(() => done(new Error('expected error to be thrown')))
-        .catch(err => {
-          expect(err.message).to.equal(
-            'Error transforming xml. xsltproc returned code "100", and signal "undefined". ' +
-            'xsltproc stderr output:\nsome error'
-          );
-          done();
-        });
+    it('errors if child process errors', async () => {
+      try {
+        await runTest('simple', spawned, 'some error');
+        assert.fail('expected error to be thrown');
+      } catch (err) {
+        expect(err.message).to.equal(
+          'Error transforming xml. xsltproc returned code "100", and signal "undefined". ' +
+          'xsltproc stderr output:\nsome error'
+        );
+      }
+    });
+
+    it('errors if xsltproc command not found', async () => {
+      try {
+        const writeErr = new Error('Error: write EPIPE');
+        writeErr.code = 'EPIPE';
+        const spawnedEpipe = {
```

Comment:
what do you think of this:
```js
const spawnedUnknownWriteErr = { 
  ...spawned,  
  stdin: { 
    ...spawned.stdin, 
    write: sinon.stub().throws('mystery error') 
  }
};
```

is it allowed in the api's the node versions?

Reply:
Turns out is not supported I guess for the reason you mention: https://github.com/medic/cht-core/runs/3616380791?check_suite_focus=true ... rolling back.

Reply:
aww yeah, I think we're a bit behind in node versions support.